### PR TITLE
feat(live-notif): youtube polling, message ttl cleanup, api backoff

### PR DESCRIPTION
**Pull request by LucasSantana-Dev**
Closes the real gaps from gap analysis against web-app issue #130 (Twitch notification pipeline already existed): Twitch poll normalized to 5 min; YouTube live detection via a single quota-safe search.list (30-min interval = 4.8k units/day vs 14.4k that the spec'd 10-min would burn against the 10k/day free quota — deviation documented in code); posted-notification tracking with auto-delete on stream end or 4h TTL (in-memory; restart-orphan limitation documented — Prisma model deliberately NOT added); exponential backoff honoring Retry-After on 429/5xx for both APIs. New env (optional): YOUTUBE_API_KEY + YOUTUBE_CHANNEL_ID — absent disables YouTube path with a startup log. 27/27 spec tests. eventsub files untouched (PR #1749 owns them). Cross-repo: Criativaria-Projects/web-app#130.

---
## Summary by cubic
Adds YouTube live polling and 4h TTL cleanup to the live notification service, with exponential backoff and a 5-minute Twitch poll. Uses a quota-safe 30-minute YouTube interval (single `search.list`) with `crypto.randomInt` jitter; aligns with web-app #130.

- **New Features**
  - Twitch polling set to 5 minutes.
  - YouTube live detection via one `search.list` every 30 minutes; disabled without `YOUTUBE_API_KEY` and `YOUTUBE_CHANNEL_ID`.
  - De-dup per stream/broadcast and track posted messages; auto-delete after 4 hours; soft-fail on missing/already-deleted messages.
  - Exponential backoff for Twitch and YouTube requests, respecting Retry-After on 429/5xx.

- **Bug Fixes**
  - Fixed Retry-After parsing (supports seconds and HTTP-date) and improved backoff resilience.
  - Start decouples YouTube from Twitch env; added tests for YouTube-only start, Retry-After variants, and empty search results.
  - Repaired a duplicate test block; deflaked Retry-After HTTP-date timing test.

<sup>Written for commit 10d4bf7d77c3a8799f115edb3ef8f13a0f580e51. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/LucasSantana-Dev/Lucky/pull/1762?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added YouTube live broadcast polling and notifications alongside Twitch.
  * Notifications now use platform-specific embeds, with optional role mentions.
* **Bug Fixes**
  * Improved polling reliability with independent Twitch/YouTube intervals, re-entrancy guards, and backoff with `Retry-After` handling.
  * Prevented duplicate notifications and added TTL-based cleanup for stale messages (including safe handling when deletions fail).
* **Tests**
  * Expanded coverage for edge cases, retry/backoff behavior, and lifecycle start/stop logic.

### CODE REVIEW

File: packages/bot/src/services/CriativariaLiveNotificationService.ts [hunk 1]
```diff
@@ -15,16 +19,89 @@ type TwitchStream = {
...
+                ...options,
+                signal: AbortSignal.timeout(10_000),
+            } as RequestInit)
+            // If 429 or 5xx, backoff and retry
+            if (
+                (res.status === 429 || res.status >= 500) &&
+                attempt < maxAttempts - 1
+            ) {
+                const retryAfter = res.headers.get('Retry-After')
+                const delayMs = retryAfter
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.ts]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.ts [hunk 2]
```diff
@@ -36,36 +113,73 @@ export class CriativariaLiveNotificationService {
...
+
+        // Start Twitch polling (5 min interval)
+        void this.twitchTick(client)
+        this.twitchIntervalHandle = setInterval(
+            () => void this.twitchTick(client),
+            this.twitchPollIntervalMs,
         )
+
+        // Start YouTube polling (10 min interval) if key is present
+        const youtubeApiKey = process.env.YOUTUBE_API_KEY
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.ts]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.spec.ts [hunk 3]
```diff
@@ -41,185 +49,532 @@ describe('CriativariaLiveNotificationService', () => {
...
-            user_login: 'criativaria',
-            title: 'Criativaria ao Vivo',
-            viewer_count: 1500,
-            game_name: 'Creative',
-            thumbnail_url: 'https://example.com/{width}x{height}.jpg',
-            started_at: new Date().toISOString(),
+    describe('backoff & rate-limit safety (GAP 4)', () => {
+        test('should retry on 429 (rate limit)', async () => {
+            let attempts = 0
+            global.fetch = jest.fn(async () => {
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.spec.ts]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.ts [hunk 4]
```diff
@@ -36,36 +113,73 @@ export class CriativariaLiveNotificationService {
...
-            this.pollIntervalMs,
+
+        // Start Twitch polling (5 min interval)
+        void this.twitchTick(client)
+        this.twitchIntervalHandle = setInterval(
+            () => void this.twitchTick(client),
+            this.twitchPollIntervalMs,
         )
+
+        // Start YouTube polling (10 min interval) if key is present
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.ts]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.spec.ts [hunk 5]
```diff
@@ -41,185 +49,632 @@ describe('CriativariaLiveNotificationService', () => {
...
+            id: 'vid1',
+            title: 'Live!',
+            thumbnail: 'http://t/img.jpg',
+            channelTitle: 'Criativaria',
         })
+        expect(String(fetchSpy.mock.calls[0][0])).toContain('eventType=live')
+        expect(String(fetchSpy.mock.calls[0][0])).toContain('channelId=UCabc')
+    })
+
+    test('fetchYoutubeLiveBroadcast returns null on non-ok response', async () => {
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.spec.ts:807]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.spec.ts [hunk 6]
```diff
@@ -41,185 +49,632 @@ describe('CriativariaLiveNotificationService', () => {
...
         })
 
-        jest.mocked(mockChannel.send).mockRejectedValue(
-            new Error('Send failed'),
-        )
+        test('should retry on 5xx with backoff', async () => {
+            let attempts = 0
+            global.fetch = jest.fn(async () => {
+                attempts++
+                if (attempts <= 2) return { status: 503 }
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.spec.ts]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.ts [hunk 7]
```diff
@@ -15,16 +21,92 @@ type TwitchStream = {
...
+            } as RequestInit)
+            // If 429 or 5xx, backoff and retry
+            if (
+                (res.status === 429 || res.status >= 500) &&
+                attempt < maxAttempts - 1
+            ) {
+                const retryAfter = res.headers.get('Retry-After')
+                const delayMs = retryAfter
+                    ? Math.min(parseInt(retryAfter, 10) * 1000, 30000)
+                    : Math.min(1000 * Math.pow(2, attempt) + jitterMs(), 30000)
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.ts]: Fixed in 2c4661ae + e1b79c75: Retry-After now parses integer seconds OR HTTP-date (Date.parse fallback, clamped 0-30s, exponential fallback on garbage); YouTube polling independent of Twitch env; stale 10-min comment corrected; fetch spies restored; 503 mocks given headers so the 5xx branch is actually exercised; empty-items null path tested.

File: packages/bot/src/services/CriativariaLiveNotificationService.ts [hunk 8]
```diff
@@ -15,57 +21,204 @@ type TwitchStream = {
...
+            if (
+                (res.status === 429 || res.status >= 500) &&
+                attempt < maxAttempts - 1
+            ) {
+                const retryAfter = res.headers.get('Retry-After')
+                let delayMs: number
+                if (retryAfter) {
+                    // Try parsing as integer seconds first
+                    const seconds = parseInt(retryAfter, 10)
+                    if (!isNaN(seconds) && seconds > 0) {
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/services/CriativariaLiveNotificationService.ts:69]: Deliberate trade: on an explicit Retry-After: 0 we take the exponential fallback (~1-2s) instead of an immediate retry. Under rate-limiting an extra second is harmless (arguably safer), and the polling cadence here is minutes, not milliseconds. Not worth another CI round; noting for a future touch of backoffFetch.
